compiler/rustc_lint/src/internal.rs

```diff
@@ -17,6 +17,10 @@ use rustc_span::symbol::{kw, sym, Symbol};
 use rustc_span::Span;
 
 declare_tool_lint! {
+    /// The `default_hash_type` lint declares use of [`std::collections::HashMap`]/[`std::collections::HashSet`],
+    /// suggesting the use of `FxHashMap`/`FxHashSet`.
+    ///
+    /// This can help as `FxHasher` can perform better than the default hasher.
```

Comment:
Maybe add that rustc doesn't care about DOS protection since input is assumed to be trusted?